tla/Network.tla

```diff
@@ -0,0 +1,86 @@
+-------------------------------- MODULE Network -------------------------------
+EXTENDS ccfraft, Sequences
+
+----------------------------------------------------------------------------------
+\* Reordering and duplication of messages:
+
+ReorderDupInitMessageVar ==
+    messages = <<>>
+    
+ReorderDupWithMessage(m, msgs) == 
+    IF m \notin (DOMAIN msgs) THEN
+        msgs @@ (m :> 1)
+    ELSE
+        [ msgs EXCEPT ![m] = @ + 1 ]
+
+ReorderDupWithoutMessage(m, msgs) == 
+    IF msgs[m] = 1 THEN
+        [ msg \in ((DOMAIN msgs) \ {m}) |-> msgs[msg] ]
+    ELSE
+        [ msgs EXCEPT ![m] = @ - 1 ]
+
+ReorderDupMessages ==
+    DOMAIN messages
+
+ReorderDupMessagesOf(s) ==
+    { m \in Messages : m.dest = s }
+
+----------------------------------------------------------------------------------
+\* Reordering and deduplication of messages (iff the spec removes message m from
+\* msgs after receiving m, i.e., ReorderNoDupWithoutMessage.)
+
+ReorderNoDupInitMessageVar ==
+    messages = {}
+
+ReorderNoDupWithMessage(m, msgs) == 
+    msgs \union {m}
+
+ReorderNoDupWithoutMessage(m, msgs) == 
+    msgs \ {m}
+
+ReorderNoDupMessages ==
+    messages
+
+ReorderNoDupMessagesOf(s) ==
+    { m \in Messages : m.dest = s }
+
+----------------------------------------------------------------------------------
+\* Ordering and duplication of messages:
+
+OrderInitMessageVar ==
```

Comment:
`DebugInvAllMessagesProcessable` currently uses `Len(Messages)` to test if the Messages is empty. Maybe an `IsEmpty` operator could replace this use?

Reply:
`Len(Messages)` doesn't make sense anyway.  Corrected to `Messages # {} ~> Messages = {}`.